File: src/main/java/com/github/joelws/release/tracker/dao/artifact/ArtifactDao.java
package com.github.joelws.release.tracker.dao.artifact;

import com.github.joelws.release.tracker.dao.GenericDao;
import com.github.joelws.release.tracker.entity.artifact.Artifact;
import com.github.joelws.release.tracker.entity.artifact.ArtifactPK;

import java.util.List;

public interface ArtifactDao extends GenericDao<Artifact, ArtifactPK>
{
    List<Artifact> list();
}

File: src/main/kotlin/com/github/joelws/release/tracker/dao/artifact/ArtifactDao.kt
package com.github.joelws.release.tracker.dao.artifact

import com.github.joelws.release.tracker.dao.GenericDao
import com.github.joelws.release.tracker.entity.artifact.Artifact
import com.github.joelws.release.tracker.entity.artifact.ArtifactPK

interface ArtifactDao : GenericDao<Artifact, ArtifactPK> {
    fun list(): List<Artifact>
}
